src/services/IFileSystem.hh

```diff
@@ -35,6 +37,25 @@ public:
     /// <returns><c>true</c> if successful, <c>false</c> if not.</returns>
     virtual bool CreateDirectory(const std::wstring& sDirectory) const = 0;
 
+    /// <summary>
+    /// Gets the size of the file (in bytes).
+    /// </summary>
+    /// <remarks>Returns -1 if the file does not exist.</remarks>
+    virtual int64_t GetFileSize(const std::wstring& sPath) const = 0;
```

Comment:
What you have is fine but just wanted to comment on this.

I have a function similar to this in `ra_utility.h` that attempts to get the file size at compile-time, should we just keep that one as an alternative or get rid of it? At worst it's constant-time, as the value is retrieved on construction. It also supports any character type specified the `ra::is_char` `type_trait `constraint.

It looks like this:
```
template<typename CharT, class = std::enable_if_t<is_char_v<CharT>>> _NODISCARD _CONSTANT_FN
filesize(_In_ const std::basic_string<CharT>& filename) noexcept
{
    using file_type = std::basic_fstream<CharT>;
    file_type file{ filename, std::ios::in | std::ios::ate | std::ios::binary };
    return file.tellg();
} // end function filesize
```
`file.tellg()` will also return -1 on error. Auto deduced return type is `std::basic_fstream<CharT>::pos_type` which is `std::streamoff` which is implementation defined but usually `long long.`

There's also an` rvalue reference` version if the string is a temporary or no longer needs to be used. I guess the main difference is that mine is assumed not to fail as only RA filenames should be passed to it.

`tellg `can throw `std::ios_base::failur`e but is usually handled internally.